ci/check_diff.sh

```diff
@@ -155,16 +177,107 @@ function check_repo() {
     STATUSES+=($?)
     set -e
 
-    echo "removing tmp_dir $tmp_dir"
+    echo -e "removing tmp_dir $tmp_dir\n\n"
     rm -rf $tmp_dir
     cd $WORKDIR
 }
 
+function write_readme() {
+    rustfmt_diff=\`rustfmt_diff.txt\`
+    feature_diff=\`feature_diff.txt\`
+    diff_file=\`diff.txt\`
+    diff_files=\`*_diff.txt\`
+
+    if [ -n "$OPTIONAL_RUSTFMT_CONFIGS" ]; then
+        OPTIONAL_CONFIG_DETAILS="* diff check optional configs: \`$OPTIONAL_RUSTFMT_CONFIGS\`"
+    fi
+
+    cat > README.md << EOL
+# Diff Check
+
+## Summary
+
+The Diff Check Job is used to validate rustfmts backwards compatability guarantees
+by running the latest rustfmt from [rust-lang/rustfmt](https://github.com/rust-lang/rustfmt) and
+comparing the formatting results against a fork or feature branch of rustfmt --
+often before deciding to merge those changes into rustfmt via a pull request.
+
+**cargo details**
+* version: \`$CARGO_VERSON\`
+
+**rustfmt details**
+* version: \`$RUSTFMT_VERSION\`
+
+**fork details**
+* repo url: $REMOTE_REPO
+* feature branch: \`$FEATURE_BRANCH\`
+* version: \`$FEATURE_VERSION\`
+$OPTIONAL_CONFIG_DETAILS
+
+## How to interpret results
+
+Diffs created by running the rustfmt binary are reported in $rustfmt_diff, and
+diffs created by running the forked rustfmt binary are stored in $feature_diff.
+The presence of $rustfmt_diff and $feature_diff are not indicative of any errors.
+Some of the real world projects that rustfmt is tested against may not use rustfmt at all.
+All the $diff_files files show is that using rustfmt on a given project would change some formatting.
+
+If a $diff_file file is present for a given project then that indicates a failure to
```

Comment:
Perhaps worth mention the version=Two / style_edition=vNext* is a scenario where formatting changes might actually be expected?

Or are we only running diffs against repositories that we know aren't using a vNext Style Edition?